### PR TITLE
feat(cli): add `--check` flag

### LINKED ISSUE #112: Flag for non-0 exit code if formatting changed the file
Hey there !

Would it be possible to add a couple of flags so that `format-package` could be executed as a pre-commit **test**? I saw it is possible to use lint-staged, but I have a pre-commit hook that checks if everything is OK and I'd like to check package.json too

This would be like this:

```sh
format-package [--check-formated,-c]
```

If the serialization of the formatted package.json is different to the actual file, this would return a non-0 exit code.

Thanks !

### PR DESCRIPTION
**Pull request by GerkinDev**
Closes #112

### CODE REVIEW

**Review by camacho — APPROVED** [review]
Generally looks good 👍

File: src/cli/index.test.ts [hunk 1]
```diff
@@ -6,13 +7,18 @@ import * as cli from '.';
...
+jest.mock('../lib', () => ({ default: mockFormat }));
 
 describe('cli', () => {
   let mockReadJSONSync;
   let mockWriteFileSync;
   let mockConsoleLog;
   let mockConsoleWarn;
 
+  beforeEach(() => {
+    mockFormat.mockImplementation((v) => v);
```
[the remark this answers is not in the document]
Comment (GerkinDev) [reply src/cli/index.test.ts:19]: Yes, it should be possible. But I don't have time to update this PR.

**Comment by GerkinDev** [comment]
Hi,

What do you need for this to be merged ?

**Comment by GerkinDev** [comment]
Ping

**Pull request merged**